### PR TITLE
fix issue #73 - Failure to compile when assigning a `const(bool)` to a node

### PR DESCRIPTION
**Pull request by BBasile**
I've fixed the issue like a barbarian so  review carefully.

### CODE REVIEW

**Comment by ZombineDev** [comment]
@BBasile can you add a test?

**Comment by BBasile** [comment]
added test + rebased.

**Comment by ZombineDev** [comment]
Thanks @BBasile!

**Comment by wilzbach** [comment]
Weird this shouldn't have passed the testsuite, the log even says:

```
D:YAML Constructor unittest
.....Failure ......................
core.exception.AssertError
core.exception.AssertError@source/dyaml/testconstructor.d(432): Assertion failure
----------------
??:? _d_assertp [0x73a1c9]
source/dyaml/testconstructor.d:432 @safe int dyaml.testconstructor.testConstructor(bool, immutable(char)[], immutable(char)[]).__foreachbody5(ref dyaml.node.Node) [0x7307c9]
source/dyaml/loader.d:304 @trusted int dyaml.loader.Loader.opApply(int delegate(ref dyaml.node.Node)) [0x6e0aba]
source/dyaml/testconstructor.d:413 void dyaml.testconstructor.testConstructor(bool, immutable(char)[], immutable(char)[]) [0x73057e]
source/dyaml/testcommon.d:149 std.typecons.__T5TupleTAyaVAyaa4_6e616d65TAAyaVAyaa9_66696c656e616d6573TE5dyaml10testcommon10TestStatusVAyaa4_6b696e64TAyaVAyaa4_696e666fZ.Tuple dyaml.testcommon.execute!(immutable(char)[], immutable(char)[]).execute(const(immutable(char)[]), void function(bool, immutable(char)[], immutable(char)[])*, immutable(char)[][], const(bool)) [0x725329]
source/dyaml/testcommon.d:57 int dyaml.testcommon.run!(immutable(char)[], immutable(char)[]).run(immutable(char)[], void function(bool, immutable(char)[], immutable(char)[])*, immutable(char)[][], immutable(char)[][]).__foreachbody5(ref immutable(char)[], ref immutable(char)[][]) [0x724e45]
??:? _aaApply2 [0x73d978]
source/dyaml/testcommon.d:44 void dyaml.testcommon.run!(immutable(char)[], immutable(char)[]).run(immutable(char)[], void function(bool, immutable(char)[], immutable(char)[])*, immutable(char)[][], immutable(char)[][]) [0x724a4c]
source/dyaml/testconstructor.d:443 void dyaml.testconstructor.__unittestL440_59() [0x7308aa]
??:? void dyaml.testconstructor.__modtest() [0x730fb4]
??:? int core.runtime.runModuleUnitTests().__foreachbody2(object.ModuleInfo*) [0x76ad0c]
??:? int object.ModuleInfo.opApply(scope int delegate(object.ModuleInfo*)).__lambda2(immutable(object.ModuleInfo*)) [0x7398e6]
??:? int rt.minfo.moduleinfos_apply(scope int delegate(immutable(object.ModuleInfo*))).__foreachbody2(ref rt.sections_elf_shared.DSO) [0x74397d]
??:? int rt.sections_elf_shared.DSO.opApply(scope int delegate(ref rt.sections_elf_shared.DSO)) [0x743b80]
??:? int rt.minfo.moduleinfos_apply(scope int delegate(immutable(object.ModuleInfo*))) [0x743909]
??:? int object.ModuleInfo.opApply(scope int delegate(object.ModuleInfo*)) [0x7398bd]
??:? runModuleUnitTests [0x76abf0]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x73e6a6]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x73e63f]
??:? _d_run_main [0x73e5af]
??:? main [0x61f259]
??:? __libc_start_main [0x15528f44]
```

File: source/dyaml/testconstructor.d [hunk 1]
```diff
@@ -83,10 +83,18 @@ Node[] constructBinary()
 
 Node[] constructBool()
 {
+    const(bool) a = true;
+    immutable(bool) b = true;
+    const bool aa = true;
+    immutable bool bb = true;
```
Comment (wilzbach) [inline source/dyaml/testconstructor.d]: There isn't a difference between `const(bool)` and `const bool`

**Comment by wilzbach** [comment]
https://github.com/dlang-community/D-YAML/pull/84 now has to fix the testsuite :/

**Pull request merged**